### PR TITLE
fix: run `build:docker` unless running staging tests

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Review by lluiscampos — APPROVED** [review]
It looks good.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -262,16 +262,25 @@ build:docker:
...
+    - when: on_success
   variables:
     DOCKER_REPOSITORY: mendersoftware/mender-client-docker-addons
     DOCKER_DIR: extra/mender-client-docker-addons
 
 publish:image:
+  rules:
+    - if: $RUN_TESTS_STAGING == "true"
+      when: never
+    - when: on_success
```
Comment (lluiscampos) [inline .gitlab-ci.yml:274]: The fallback rule here (and below) should be [this line](https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-docker-build.yml#L95),

**Review by lluiscampos — CHANGES_REQUESTED** [review]
The publish images cannot be run always but only on special branches. See comment below.

File: .gitlab-ci.yml (hunk 1 above)
Comment (tranchitella) [reply .gitlab-ci.yml]: Force-pushed

**Review by lluiscampos — APPROVED** [review]

**Comment by tranchitella** [comment]
Merging despite the failure, which is unrelated to this PR.

**Pull request merged**